projects/graph/src/graph.py

```diff
@@ -5,5 +5,93 @@
 
```

Comment:
Great job so far Patrick ! I know you mentioned that you were still working on the recursion part , remember to reach out if you need help!